core/src/test/scala/unit/kafka/server/DescribeGroupsRequestTest.scala

```diff
@@ -0,0 +1,118 @@
+  @ClusterTest(serverProperties = Array(
+    new ClusterConfigProperty(key = "unstable.api.versions.enable", value = "true"),
+    new ClusterConfigProperty(key = "group.coordinator.new.enable", value = "true"),
+    new ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "1"),
+    new ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "1")
+  ))
+  def testDescribeGroupsWithOldConsumerGroupProtocolAndNewGroupCoordinator(): Unit = {
+    testDescribeGroups()
+  }
+
+  @ClusterTest(clusterType = Type.ALL, serverProperties = Array(
+    new ClusterConfigProperty(key = "unstable.api.versions.enable", value = "false"),
+    new ClusterConfigProperty(key = "group.coordinator.new.enable", value = "false"),
+    new ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "1"),
+    new ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "1")
+  ))
+  def testDescribeGroupsWithOldConsumerGroupProtocolAndOldGroupCoordinator(): Unit = {
+    testDescribeGroups()
+  }
+
+  private def testDescribeGroups(): Unit = {
+
+    // Creates the __consumer_offsets topics because it won't be created automatically
+    // in this test because it does not use FindCoordinator API.
+    createOffsetsTopic()
+
+    // Create the topic.
+    createTopic(
+      topic = "foo",
+      numPartitions = 3
+    )
+
+    // Join the consumer group. Note that we don't heartbeat here so we must use
+    // a session long enough for the duration of the test.
+    val (memberId1, _) = joinConsumerGroupWithOldProtocol(
+      groupId = "grp-1",
+      metadata = Array(1, 2, 3)
+    )
+    val (memberId2, _) = joinConsumerGroupWithOldProtocol(
+      groupId = "grp-2",
+      metadata = Array(1, 2, 3),
+      completeRebalance = false
+    )
+
+    for (version <- ApiKeys.DESCRIBE_GROUPS.oldestVersion() to ApiKeys.DESCRIBE_GROUPS.latestVersion(isUnstableApiEnabled)) {
+      assertEquals(
+        List(
+          new DescribedGroup()
+            .setGroupId("grp-1")
+            .setGroupState(GenericGroupState.STABLE.toString)
+            .setProtocolType("consumer")
+            .setProtocolData("consumer-range")
+            .setMembers(List(
+              new DescribedGroupMember()
+                .setMemberId(memberId1)
+                .setGroupInstanceId(null)
+                .setClientId("client-id")
+                .setClientHost("/127.0.0.1")
+                .setMemberMetadata(Array(1, 2, 3))
+            ).asJava),
+          new DescribedGroup()
+            .setGroupId("grp-2")
+            .setGroupState(GenericGroupState.COMPLETING_REBALANCE.toString)
+            .setProtocolType("consumer")
+            .setMembers(List(
+              new DescribedGroupMember()
+                .setMemberId(memberId2)
+                .setGroupInstanceId(null)
+                .setClientId("client-id")
+                .setClientHost("/127.0.0.1")
+            ).asJava),
+          new DescribedGroup()
```

Comment:
nit: Should we add a comment for this case? It is obvious that we should return a Dead group when the group does not exist.